src/components/CheckboxWithLabel.js

```diff
@@ -73,55 +78,55 @@ const defaultProps = {
     forwardedRef: () => {},
 };
 
-class CheckboxWithLabel extends React.Component {
-    constructor(props) {
-        super(props);
-
-        // We need to pick the first value that is strictly a boolean
-        // https://github.com/Expensify/App/issues/16885#issuecomment-1520846065
-        this.isChecked = _.find([props.value, props.defaultValue, props.isChecked], (value) => _.isBoolean(value));
-
-        this.LabelComponent = props.LabelComponent;
-
-        this.toggleCheckbox = this.toggleCheckbox.bind(this);
-    }
-
-    toggleCheckbox() {
-        this.props.onInputChange(!this.isChecked);
-        this.isChecked = !this.isChecked;
-    }
-
-    render() {
-        return (
-            <View style={this.props.style}>
-                <View style={[styles.flexRow, styles.alignItemsCenter, styles.breakWord]}>
-                    <Checkbox
-                        isChecked={this.isChecked}
-                        onPress={this.toggleCheckbox}
-                        label={this.props.label}
-                        hasError={Boolean(this.props.errorText)}
-                        forwardedRef={this.props.forwardedRef}
-                    />
-                    <PressableWithFeedback
-                        focusable={false}
-                        wrapperStyle={[styles.flex1]}
-                        onPress={this.toggleCheckbox}
-                        pressDimmingValue={variables.checkboxLabelActiveOpacity}
-                        hoverDimmingValue={variables.checkboxLabelHoverOpacity}
-                        style={[styles.ml3, styles.pr2, styles.w100, styles.flexRow, styles.flexWrap, styles.flexShrink1, styles.alignItemsCenter, styles.noSelect]}
-                    >
-                        {this.props.label && <Text style={[styles.ml1]}>{this.props.label}</Text>}
-                        {this.LabelComponent && <this.LabelComponent />}
-                    </PressableWithFeedback>
-                </View>
-                <FormHelpMessage message={this.props.errorText} />
+const CheckboxWithLabel = (props) => {
+    // We need to pick the first value that is strictly a boolean
+    // https://github.com/Expensify/App/issues/16885#issuecomment-1520846065
+    const [isChecked, setIsChecked] = useState(_.find([props.value, props.defaultValue, props.isChecked], (value) => _.isBoolean(value)));
```

Comment:
We also need to check the case when the checkbox is controlled or uncontrolled.

More details here:

https://github.com/Expensify/App/issues/73894#issuecomment-3470702134